src/main/java/org/sonarlint/intellij/finding/persistence/FindingsManager.java

```diff
@@ -0,0 +1,247 @@
+import org.sonarlint.intellij.finding.tracking.Tracker;
+import org.sonarlint.intellij.finding.tracking.Tracking;
+import org.sonarlint.intellij.finding.issue.LiveIssue;
+import org.sonarlint.intellij.messages.IssueStoreListener;
+import org.sonarsource.sonarlint.core.commons.IssueSeverity;
+
+import static java.util.Collections.emptyList;
+
+/**
+ * Stores findings associated to a {@link PsiElement}, {@link RangeMarker} or  {@link PsiFile}.
+ * Findings are either issues or hotspots.
+ * Findings can then be displayed as annotations at any time.
+ */
+@ThreadSafe
+public class FindingsManager {
+  private final Project myProject;
+  private final LiveFindingCache<LiveIssue> liveIssueCache;
+  private final LiveFindingCache<LiveSecurityHotspot> liveSecurityHotspotCache;
+
+  public FindingsManager(Project project) {
+    this(project, new LiveFindingCache<>(project, new FindingPersistence<>(project, "issue")));
+  }
+
+  @NonInjectable
+  FindingsManager(Project project, LiveFindingCache<LiveIssue> liveIssueCache) {
+    myProject = project;
+    this.liveIssueCache = liveIssueCache;
+    this.liveSecurityHotspotCache = new LiveFindingCache<>(project, new FindingPersistence<>(project, "securityhotspot"));
+    project.getMessageBus().connect().subscribe(ProjectManager.TOPIC, new ProjectManagerListener() {
+      @Override
+      public void projectClosing(@NotNull Project project) {
+        try {
+          if (project == myProject) {
+            // Flush findings before project is closed, because we need to resolve module paths to compute the key
+            liveIssueCache.flushAll();
+            liveSecurityHotspotCache.flushAll();
+          }
+        } catch (Exception e) {
+          SonarLintConsole.get(myProject).error("Cannot flush issues", e);
+        }
+      }
+    });
+  }
+
+  public void clearAllIssuesForAllFiles() {
+    liveIssueCache.clear();
+    myProject.getMessageBus().syncPublisher(IssueStoreListener.SONARLINT_ISSUE_STORE_TOPIC).allChanged();
+  }
+
+  public void clearAllFindingsForAllFiles() {
+    liveIssueCache.clear();
+    liveSecurityHotspotCache.clear();
+    myProject.getMessageBus().syncPublisher(IssueStoreListener.SONARLINT_ISSUE_STORE_TOPIC).allChanged();
+  }
+
+  public void clearAllFindingsForFiles(Collection<VirtualFile> files) {
+    files.forEach(liveIssueCache::clear);
+    files.forEach(liveSecurityHotspotCache::clear);
+    myProject.getMessageBus().syncPublisher(IssueStoreListener.SONARLINT_ISSUE_STORE_TOPIC).filesChanged(new HashSet<>(files));
+  }
+
+  public void removeResolvedFindings(VirtualFile file, List<LiveIssue> issues, List<LiveSecurityHotspot> securityHotspots) {
+    liveIssueCache.removeFindings(file, issues);
+    liveSecurityHotspotCache.removeFindings(file, securityHotspots);
+    notifyFindingsChangedForFile(file);
+  }
+
+  public boolean neverAnalyzed(VirtualFile file) {
+    return liveIssueCache.getLive(file) == null;
```

Comment:
I think we should also check `&& liveSecurityHotspotCache.getLive(file) == null`

Reply:
Good catch! And I see we have a `neverAnalyzed` and `wasAnalyzed`, I will get rid of one of them

Reply:
In fact the meaning of both methods is slightly different so I will keep them both but fix the issue and rename one